sdk/servicebus/Azure.Messaging.ServiceBus/src/Receiver/ServiceBusReceiver.cs

```diff
@@ -238,6 +253,28 @@ public class ServiceBusReceiver : IAsyncDisposable
             return null;
         }
 
+        private async Task ApplyPlugins(IList<ServiceBusReceivedMessage> messages)
+        {
+            foreach (ServiceBusPlugin plugin in _plugins)
+            {
+                string pluginType = plugin.GetType().Name;
```

Comment:
Would it make sense to either

a) Make the plugin name an explicit overrideable concept on the plugin class that by default uses a cached instance of `plugin.GetType().Name` but can be overridden by plugin authors _or_
b) Given that the plugin name is only relevant for the path where the logger `IsEnabled` pass the plugin to the logger and then use `plugin.GetType().Name` within the logger _or_
c) check logger `IsEnabled` on this level and then assign `string pluginType = plugin.GetType().Name;` and use it.

Any thoughts?